src/main/java/de/qabel/desktop/DesktopClient.java

```diff
@@ -149,31 +209,28 @@ public void start(Stage stage) throws Exception {
         scene = new Scene(new LoginView().getView(), 370, 570, true, aa);
         primaryStage.setScene(scene);
 
-        config.addObserver((o, arg) -> {
+        config.onSetAccount(account ->
             Platform.runLater(() -> {
-                if (arg instanceof Account) {
-                    try {
-                        new Thread(getSyncDaemon(config)).start();
-                        new Thread(getDropDaemon(config)).start();
-                        view = new LayoutView();
-                        Parent view = this.view.getView();
-                        Scene layoutScene = new Scene(view, 900, 600, true, aa);
-                        Platform.runLater(() -> primaryStage.setScene(layoutScene));
-
-                        if (config.getSelectedIdentity() != null) {
-                            addShareMessageRenderer(config.getSelectedIdentity());
-                        }
-                    } catch (Exception e) {
-                        logger.error("failed to init background services: " + e.getMessage(), e);
-                        //TODO to something with the fault
+                try {
+                    new Thread(getSyncDaemon(getBoxSyncConfigRepository())).start();
+                    new Thread(getDropDaemon(config)).start();
+                    view = new LayoutView();
+                    Parent view = this.view.getView();
+                    Scene layoutScene = new Scene(view, 900, 600, true, aa);
+                    Platform.runLater(() -> primaryStage.setScene(layoutScene));
+
+                    if (config.getSelectedIdentity() != null) {
+                        addShareMessageRenderer(config.getSelectedIdentity());
                     }
-                } else if (arg instanceof Identity) {
-                    addShareMessageRenderer((Identity) arg);
+                } catch (Exception e) {
+                    logger.error("failed to init background services: " + e.getMessage(), e);
+                    //TODO to something with the fault
```

Comment:
Yup. Maybe also sending the stracktrace in a crash report because that would be a pretty significant crash.